### DISCUSSION TITLE
Lowering Monitoring Mode Interval

### DISCUSSION THREAD

**Discussion by BlueBell-XA** [abraunegg/onedrive#3043] [General]
Hi @abraunegg 

Firstly, excellent work on not only your incredible work, but the documentation you've provided. It's been invaluable to me to get a single Folder in SharePoint monitored for download-only.

I would like to know however, why there is a hard lower limit of 300 seconds for the 'monitor_interval' within the onedrive config?
I can change this value to, say, "350", and when I perform `--display-config`, the change is reflected there.
If I change it to anything lower than 350 (I was hoping for 30 seconds), the value is overridden and interval is set back to 300.

Whether it helps or not... I'm running the following:
- Ubuntu 24.04 Noble
- Onedrive 2.5.3
- Curl 8.11.0

**Comment by abraunegg** [comment]
@BlueBell-XA 
The easiest answer is this - what is your online rate of data change?

By setting this value to something lower, like 30 seconds, the client with be *constantly* scanning your online data via the API.

This will:
1. Cause a significant increase of API access usage for you, and give rise to more 429 response events
2. Cause excessive amd constant CPU load on your system

If your data is absolutely changing every 30 seconds online, configure a webhook to get real time interactions, before the next full sync is done after the minimum of 300 seconds.

The enforced minimum value is there to ensure your system operates in the most reliable manner.

**Reply by BlueBell-XA** [reply]
Thanks for the prompt reply!

That makes total sense. I mainly wanted to ensure it was an intended feature.

As for usage... it's running on a tablet, whos sole purpose is to act as a 3D printing hub... The tablet will send print files to specific printers and is essentially our solution to version control. The files won't change very often, however when they do, the user would just need to wait around for 5 minutes before making use of a new file.

Upon reflecting on it, this is certainly not a big waste of time for the amount of times i expect these files to update... however, if it ever does become problematic, implementing webhooks as you suggested will be the way to go.

**Comment by BlueBell-XA** [comment]
Question answered - Thanks!

**Comment by abraunegg** [comment]
When https://github.com/abraunegg/onedrive/issues/2360 is implemented, the intention is that monitor interval will be raised to 3600 .. as online changes will automatically come in without configuration of a webhook
